api/src/main/java/org/openmrs/module/appointments/service/impl/TeleconsultationAppointmentService.java

```diff
@@ -17,4 +37,49 @@ public String generateTeleconsultationLink(Appointment appointment) {
         }
         return new MessageFormat(tcServerUrl).format(new Object[] {appointment.getUuid()} );
     }
+
+    public AdhocTeleconsultationResponse generateAdhocTeleconsultationLink(String patientUuid, String provider) {
```

Comment:
I think we can use the appointment object itself, instead of creating a new class.

Reply:
@angshu Since this is not an appointment and Appointment object has lot of other fields which are not needed for this implementation having a new object with only required fields would be better I felt

Reply:
@angshu : Is it ok to close this comment?

Reply:
We decided that for now, the adhoc teleconsult will be associated with an appointment. Anyway, I see that its directly being used from the controller - please change the parameter to providerUuid .. also its not being validated whether provider exists or not.